### PR TITLE
Exceptions instead of Throwables

### PR DESCRIPTION
**Pull request by tdhd**
*Description of changes:*

Use `Exception` instead of `Throwable`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by sscdotopen** [comment]
Whats the reasoning here? I think we explictly made this catch all throwables so that no error is  propagated to the invoking code.

**Comment by tdhd** [comment]
Apart from a few corner cases `Throwable`s shouldn't be caught. When catching `Throwable`s we include `Exception` and `Error` where `Error` usually is not recoverable from, like `VirtualMachineError` or `OutOfMemoryError`, i.e. we want the invoking code to fail.

File: src/main/scala/com/amazon/deequ/analyzers/runners/MetricCalculationException.scala [hunk 1]
```diff
@@ -21,12 +21,12 @@ abstract class MetricCalculationException(message: String) extends Exception(mes
...
 
-  def this(message: String, cause: Throwable) {
+  def this(message: String, cause: Exception) {
     this(message)
     initCause(cause)
   }
 
-  def this(cause: Throwable) {
+  def this(cause: Exception) {
     this(Option(cause).map(_.toString).orNull, cause)
```
Comment (sscdotopen) [inline src/main/scala/com/amazon/deequ/analyzers/runners/MetricCalculationException.scala]: We should have better logic for the case where cause can be null. We should not pass the nulls to the constructor.

Comment (tdhd) [reply src/main/scala/com/amazon/deequ/analyzers/runners/MetricCalculationException.scala]: When are we expecting the cause to be null?

Comment (tdhd) [reply src/main/scala/com/amazon/deequ/analyzers/runners/MetricCalculationException.scala]: Yes, I was wondering why we are explicitly handling the case when `cause` can be null. Are we expecting it to be null at any point?

Comment (sscdotopen) [reply src/main/scala/com/amazon/deequ/analyzers/runners/MetricCalculationException.scala]: No, not really. We could check it and invoke the constructor with a custom runtime exception to be super-duper safe, but we should not propagate nulls.

**Pull request closed without merging**